flink-runtime-web/web-dashboard/src/app/pages/task-manager/metrics/task-manager-metrics.component.html

```diff
@@ -294,3 +294,75 @@
     </div>
   </div>
 </nz-card>
+
+<ng-template #tmResourceRow let-detail="detail" let-name="name" let-property="prop">
+  <tr>
+    <td>{{ name }}</td>
+    <td>
+      <nz-progress
+        nzSize="small"
+        nzStrokeLinecap="square"
+        [nzPercent]="
+          +(
+            (detail?.freeResource[property] / detail?.totalResource[property]) * 100
+            | number: '1.0-2'
+          )
+        "
+        nzStatus="normal"
+      ></nz-progress>
+      {{ detail?.freeResource[property] | number }} / {{ detail?.totalResource[property] | number }}
+    </td>
+  </tr>
+</ng-template>
+
+<nz-card nzTitle="Resources" nzSize="small">
+  <nz-table
+    nzBordered
+    nzTitle="Unassigned resources"
+    [nzTemplateMode]="true"
+    [nzFrontPagination]="false"
+    [nzShowPagination]="false"
+    [nzSize]="'small'"
+    class="no-border"
+    [nzWidthConfig]="['200px', null]"
+  >
+    <tbody>
```

Comment:
this would make it more consistent with the rest, but I'm not sure if it makes too much sense 🤔 
```suggestion
    <thead>
      <tr>
        <th>Type</th>
        <th>Count</th>
      </tr>
    </thead>
    <tbody>
```

Reply:
It looks better with a header row.